app/Http/Controllers/DonationTransactionController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * NOTICE OF LICENSE.
+ *
+ * UNIT3D Community Edition is open-sourced software licensed under the GNU Affero General Public License v3.0
+ * The details is bundled with this project in the file LICENSE.txt.
+ *
+ * @project    UNIT3D Community Edition
+ *
+ * @author     HDVinnie <hdinnovations@protonmail.com>
+ * @license    https://www.gnu.org/licenses/agpl-3.0.en.html/ GNU Affero General Public License v3.0
+ */
+
+namespace App\Http\Controllers;
+
+use App\Http\Requests\Staff\StoreDonationTransactionRequest;
+use App\Models\DonationItem;
+use App\Models\DonationTransaction;
+use Illuminate\Support\Facades\Redirect;
+use PrevailExcel\Nowpayments\Facades\Nowpayments;
+use Exception;
+
+class DonationTransactionController extends Controller
+{
+    /**
+     * Collect Order data and create Payment.
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function createCryptoPayment(StoreDonationTransactionRequest $request)
+    {
+        $user = $request->user();
+        $price = null;
+
+        // Validate
+        $request->validated();
+
+        try {
+            $price = DonationItem::query()->select(['price_usd'])->where('id', '=', $request->item)->value('price_usd');
```

Comment:
```suggestion
            $price = DonationItem::query()->select(['price_usd'])->findOrFail($request->item)->value('price_usd');
```

Reply:
I needed to revert this, as it did not find the price and returned 0.00 always. Please see: ee9fdf897ebe89b1b3e4634bd889e46de297dadd